Source/WebCore/rendering/style/SVGRenderStyle.cpp

```diff
@@ -154,72 +175,72 @@ StyleDifference SVGRenderStyle::diff(const SVGRenderStyle& other) const
         || m_nonInheritedFlags.flagBits.alignmentBaseline != other.m_nonInheritedFlags.flagBits.alignmentBaseline
         || m_nonInheritedFlags.flagBits.dominantBaseline != other.m_nonInheritedFlags.flagBits.dominantBaseline
         || m_nonInheritedFlags.flagBits.baselineShift != other.m_nonInheritedFlags.flagBits.baselineShift)
-        return StyleDifference::Layout;
+        return true;
 
     // Text related properties influence layout.
-    bool miscNotEqual = m_miscData != other.m_miscData;
-    if (miscNotEqual && m_miscData->baselineShiftValue != other.m_miscData->baselineShiftValue)
-        return StyleDifference::Layout;
+    if (m_miscData->baselineShiftValue != other.m_miscData->baselineShiftValue)
+        return true;
 
     // The x or y properties require relayout.
     if (m_layoutData != other.m_layoutData)
-        return StyleDifference::Layout; 
+        return true; 
 
     // Some stroke properties, requires relayouts, as the cached stroke boundaries need to be recalculated.
-    if (m_strokeData != other.m_strokeData) {
-        if (m_strokeData->paintType != other.m_strokeData->paintType
-            || m_strokeData->paintColor != other.m_strokeData->paintColor
-            || m_strokeData->paintUri != other.m_strokeData->paintUri
-            || m_strokeData->dashArray != other.m_strokeData->dashArray
-            || m_strokeData->dashOffset != other.m_strokeData->dashOffset
-            || m_strokeData->visitedLinkPaintColor != other.m_strokeData->visitedLinkPaintColor
-            || m_strokeData->visitedLinkPaintUri != other.m_strokeData->visitedLinkPaintUri
-            || m_strokeData->visitedLinkPaintType != other.m_strokeData->visitedLinkPaintType)
-            return StyleDifference::Layout;
-
-        // Only the stroke-opacity case remains, where we only need a repaint.
-        ASSERT(m_strokeData->opacity != other.m_strokeData->opacity);
-        return StyleDifference::Repaint;
-    }
+    if (m_strokeData->paintType != other.m_strokeData->paintType
+        || colorDiff(m_strokeData->paintColor, other.m_strokeData->paintColor, currentColorDiffers)
+        || m_strokeData->paintUri != other.m_strokeData->paintUri
+        || m_strokeData->dashArray != other.m_strokeData->dashArray
+        || m_strokeData->dashOffset != other.m_strokeData->dashOffset
+        || colorDiff(m_strokeData->visitedLinkPaintColor, other.m_strokeData->visitedLinkPaintColor, currentColorDiffers)
+        || m_strokeData->visitedLinkPaintUri != other.m_strokeData->visitedLinkPaintUri
+        || m_strokeData->visitedLinkPaintType != other.m_strokeData->visitedLinkPaintType)
+        return true;
```

Comment:
Why does a change of paintColor require layout?

Reply:
I doubt it does actually (like opacity which is explicitly move to the repaint part) but I don't want to mix several changes in this patch.